### PR TITLE
ci: enable artifact attenstations

### PR DESCRIPTION
**Pull request by CharlieMc0**
# Description

Enables GitHub attestations to verify the binaries available on our releases page were built by our CI. 

[More info about GitHub attestations here](https://github.blog/2024-05-02-introducing-artifact-attestations-now-in-public-beta/)

Attestations will be available to view https://github.com/zeta-chain/node/attestations, as a JSON file attached to the releases page, and at https://search.sigstore.dev/. 

Closes: <PD-XXXX>

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Include instructions and any relevant details so others can reproduce. 

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions 

# Checklist:

- [ ] I have added unit tests that prove my fix feature works

### CODE REVIEW

**Comment by gartnera** [comment]
Can you do a release on https://github.com/zeta-chain/ci-testing-node to validate it's all working correctly?

**Comment by CharlieMc0** [comment]
Yes already done here for a couple of binaries - https://github.com/zeta-chain/ci-testing-node/releases/tag/v02
A new one with all binaries renabled is running now.

File: .github/workflows/publish-release.yml [hunk 1]
```diff
@@ -478,6 +482,22 @@ jobs:
...
+        uses: actions/attest-build-provenance@v1
+        with:
+          subject-path: |
+            dist/zetacored_**/*
+            dist/zetaclientd_**/*
+            dist/checksums.txt
+
+      - name: Upload Attestation Bundle
+        env: 
+          GH_TOKEN: ${{ secrets.PAT_GITHUB_SERVICE_ACCT }}
```
Comment (gartnera) [inline .github/workflows/publish-release.yml:496]: Why do we need this token rather than the default token?

**Review by gartnera — APPROVED** [review]

**Review by lumtis — APPROVED** [review]
LGTM

File: .github/workflows/publish-release.yml (hunk 1 above)
Comment (CharlieMc0) [reply .github/workflows/publish-release.yml:496]: I was thinking the same thing when working on this. Before there were some restrictions with creating release or something like that but we may not have been setting the permissions correctly. I would like to try going back to the default one. I am going to merge this PR as is and I've added another issue. 

https://github.com/zeta-chain/node/issues/2391

Comment (CharlieMc0) [reply .github/workflows/publish-release.yml]: Removed custom GH token from this workflow and left an issue to clean up the rest.

**Pull request merged**